sci-libs/rocFFT/rocFFT-5.7.1-r1.ebuild

```
@@ -29,7 +29,7 @@ perfscripts? (
 	dev-python/pandas[${PYTHON_USEDEP}] )
 ${PYTHON_DEPS}"
 
-DEPEND="dev-util/hip
+DEPEND="=dev-util/hip-5*
```

missing dependency (boost was missing already from the version 5.1.3)
```suggestion
DEPEND="=dev-util/hip-5*
	benchmark? (
		dev-libs/boost
		sci-libs/hipRAND
	)
```

---

Also, I have the following additional proposal for your `rocBLAS-5.7.1` ebuild, which I didn't know how to include directly. All these fixes are intended to resolve build errors with specific USE flags.
```patch
--- sci-libs/rocBLAS/rocBLAS-5.7.1-r1.ebuild.AngryLoki
+++ sci-libs/rocBLAS/rocBLAS-5.7.1-r2.ebuild
@@ -4,7 +4,7 @@
 EAPI=8
 
 DOCS_BUILDER="doxygen"
-DOCS_DIR="docs"
+DOCS_DIR="docs/.doxygen"
 DOCS_DEPEND="media-gfx/graphviz"
 ROCM_VERSION=${PV}
 inherit cmake docs edo multiprocessing rocm
@@ -57,6 +57,11 @@
 }
 
 src_configure() {
+	if [[ ! -d /usr/include/boost ]]; then
+		# the case of dev-cpp/msgpack-cxx[-boost] with dev-util/Tensile
+		append-cxxflags -DMSGPACK_NO_BOOST
+	fi
+
 	addpredict /dev/random
 	addpredict /dev/kfd
 	addpredict /dev/dri/
```
BTW, thank you for the new and updated packages. They work well for me, as far as I could check (also after including `boost`, of course).

---

> which I didn't know how to include directly.

You can file a PR against https://github.com/AngryLoki/gentoo/tree/rocm-5.7 . The path needs a `${EPREFIX}` prepended and you need to explain the `DOCS_DIR` change.

---

> BTW, thank you for the new and updated packages. They work well for me, as far as I could check (also after including `boost`, of course).

same here. Finally ROCm is in a workable state for me (gfx1100 user)

---

I tested all the merged ebuilds, and they work OK. The OpenCL issue is still present (of course, whole ROCm was rebuilt with the same toolchain), but the segfault is buried deep in /usr/lib64/libamdocl64.so (perhaps it is a problem of the client application called gpuowl, I didn't tested anything else; and clinfo doesn't have any problems) and USE=debug won't give more details in gdb, so let's ignore it for now (rocm-opencl-runtime-5.5.1 is working well for me).

In the meantime, I (hopefully) found a fix for an older bug in rocSPARSE, see https://bugs.gentoo.org/924446

Also, please add missing dependency in rocWMMA:
rocWMMA[test] depends on rocBLAS (during configure step, and probably also later)

I'm quite happy that fixed ROCm 5.7.1 works also on APU (Ryzen Pro 5750G). There are some freezes (maybe due to currently testing it on a text console only), but they can be easily "unfrozen" by simultaneously running rocPRIM_benchmark_warp_exchange. Of course, a dedicated GPU doesn't have such problems.